lib/redirect.js

```diff
@@ -172,48 +172,22 @@ Redirect.prototype.onResponse = function (response) {
     // remove content-length header since FormValue may be dropped if its not a file stream
     request.removeHeader('content-length')
 
-    var formData = {}
-    var resetFormData = function (key, value) {
-      var newValue
-      var formValue = value && value.hasOwnProperty('value') ? value.value : value
-
-      // if `formValue` is of type stream
-      if (typeof formValue === 'object' && typeof formValue.pipe === 'function') {
+    var formData = []
+    var resetFormData = function (key, value, paramOptions) {
+      // if `value` is of type stream
+      if (typeof value === 'object' && typeof value.pipe === 'function') {
         // bail out if not a file stream
-        if (!(formValue.hasOwnProperty('fd') && formValue.path)) return
+        if (!(value.hasOwnProperty('fd') && value.path)) return
         // create new file stream
-        formValue = fs.createReadStream(formValue.path)
-      }
-
-      if (value && value.hasOwnProperty('value') && value.hasOwnProperty('options')) {
-        newValue = { value: formValue, options: value.options }
-      } else {
-        newValue = formValue
-      }
-
-      if (!formData.hasOwnProperty(key)) {
-        formData[key] = newValue
-        return
-      }
-
-      // at this point, we know that form has duplicate
-      if (!Array.isArray(formData[key])) {
-        formData[key] = [formData[key]]
+        value = fs.createReadStream(value.path)
       }
 
-      formData[key].push(newValue)
+      formData.push({key: key, value: value, options: paramOptions})
     }
-    for (var formKey in request.formData) {
-      if (request.formData.hasOwnProperty(formKey)) {
-        var formValue = request.formData[formKey]
-        if (Array.isArray(formValue)) {
-          for (var j = 0; j < formValue.length; j++) {
-            resetFormData(formKey, formValue[j])
-          }
-        } else {
-          resetFormData(formKey, formValue)
-        }
-      }
+    for (var i = 0, ii = request.formData.length; i < ii; i++) {
+      var formParam = request.formData[i]
+      if (!formParam) { continue }
+      resetFormData(formParam.key, formParam.value, formParam.options)
```

Comment:
`formParam && ...`

Reply:
handled above, are you suggesting to follow this approach instead?
Following the existing project conventions.